stix2/sources/__init__.py

```diff
@@ -557,150 +687,198 @@ def add_filter(self, filters):
-                            break
-                    elif filter_['op'] == "in":
-                        if not stix_obj[filter_['field']] in filter_['value']:
-                            clean = False
-                            break
-                    else:
-                        # filter operation not supported
-                        continue
-
-                    # TODO: I think the rest of the operations only
-                    # apply to timestamps, in which case I don't think
-                    # simple operator usage (like below) works
-
-                    # elif filter_['op'] == ">":
-                    #     if not stix_obj[filter_['field']] > filter_['value']:
-                    #         clean = False
-                    #         break
-                    #
-                    # elif filter_['op'] == "<":
-                    #     if not stix_obj[filter_['field']] < filter_['value']:
-                    #         clean = False
-                    #         break
-                    #
-                    # elif filter_['op'] == ">=":
-                    #     if not stix_obj[filter_['field']] >= filter_['value']:
-                    #         clean = False
-                    #         break
-                    #
-                    # elif filter_['op'] == "<=":
-                    #     if not stix_obj[filter_['field']] <= filter_['value']:
-                    #         clean = False
-                    #         break
-
-                except TypeError:
-                    # type mismatch of comparison operands - ignore filter,
-                    # no error raised for now
-                    pass
-
-            # if object unmarked after all filter, add it
-            if clean:
-                filtered_stix_objs.append(stix_obj)
-
-            clean = True
-
-        return filtered_stix_objs
+        return copy.deepcopy(list(self.filters.values()))
 
     def deduplicate(self, stix_obj_list):
-        """deduplicate a list of STIX objects into a unique set
+        """deduplicate a list fo STIX objects to a unique set
 
-        reduces a set of STIX objects to unique set by looking
+        Reduces a set of STIX objects to unique set by looking
         at 'id' and 'modified' fields - as a unique object version
         is determined by the combination of those fields
 
         Args:
             stix_obj_list (list): list of STIX objects (dicts)
 
         Returns:
-            (list): a unique set of the passed STIX object list
-
-
+                (list): unique set of the passed list of STIX objects
         """
+
         unique = []
-        have = False
+        dont_have = False
         for i in stix_obj_list:
+            dont_have = False
             for j in unique:
-                if i['id'] == j['id'] and i['modified'] == j['modified']:
-                    have = True
+                for field in STIX_VERSION_FIELDS:
+                    if not i[field] == j[field]:
```

Comment:
'!=' is much nicer than 'not =='.  And does your `break` line below need an extra level of indent?

If `i` and `j` have a mismatched field, that means the objects are different.  It doesn't mean `i` isn't in the `unique` list.  It could just be in a different location, i.e. it could match a different `j`.  I think DataSource.deduplicate does it right: it breaks on a match instead of a non-match.

Looks like you switched to using `STIX_VERSION_FIELDS` here, but not in DataSource?  Seems like there are two implementations that have lost sync.  Perhaps you could reuse a single implementation.  Neither method uses `self`, so they don't even have to be methods.  Maybe make it a free function?